roles/kubernetes/preinstall/tasks/0040-set_facts.yml

```
@@ -211,7 +211,7 @@
 - name: generate nameservers for resolvconf, including cluster DNS
   set_fact:
     nameserverentries: |-
-      {{ (([nodelocaldns_ip] if enable_nodelocaldns else []) + (coredns_server|d([]) if not enable_nodelocaldns else []) + nameservers|d([]) + cloud_resolver|d([]) + (configured_nameservers|d([]) if not disable_host_nameservers|d()|bool else [])) | unique | join(',') }}
+      {{ (([nodelocaldns_ip] if enable_nodelocaldns else []) + (coredns_server|d([]) if not enable_nodelocaldns else []) + nameservers|d([]) + cloud_resolver|d([]) + ((configured_nameservers if configured_nameservers|bool else []) if not disable_host_nameservers|d()|bool else [])) | unique | join(',') }}
```

```suggestion
      {{ (([nodelocaldns_ip] if enable_nodelocaldns else []) + (coredns_server|d([]) if not enable_nodelocaldns else []) + nameservers|d([]) + cloud_resolver|d([]) + ((configured_nameservers if not(configured_nameservers|bool) else []) if not disable_host_nameservers|d()|bool else [])) | unique | join(',') }}
```

Based on your description of the issue I think you would need to add a `not` to fix this. TIL that the bool filter doesn't work like the bool cast in python: https://github.com/ansible/ansible/blob/eebfd71a6da396b011e664604c9991543205d780/lib/ansible/plugins/filter/core.py#L89, so [1,2,3]| bool == false essentially...

I don't know what else is using that though, might be nice to fix this directly in `roles/kubernetes/preinstall/tasks/0040-set_facts.yml` whenever it's defined to not have similar problem elsewhere?